### ISSUE TITLE
Ideation Part B (combined)

### ISSUE THREAD

**Issue by Pitsco** [Pitsco/nba-analysis#7]
# Overall Team Organization:
Create 3 branches: one branch for fantasy, one branch for login, one for overall stats page

![image](https://github.com/Pitsco/nba-analysis/assets/45215873/78ccaee8-2752-4784-ba33-399ebd7a6211)

We are going to make sure that for every pull request, there are at least **TWO** team members that review the commit, submit a review comment (based on whether or not it should be merged) and then move forward. 

[Planning Demonstration](https://github.com/Pitsco/nba-analysis/issues/6)

# Individual Planning: 
## Vardaan:
**FEATURE WORK ON FRONTEND & BACKEND**: 

1. I will be working on implementing the NBA API for fetching the relevant data for analysis.

2. I will also be helping on the backend to implement data analysis algorithms to process the NBA data from the API. 

PREPARED KEY COMMITS:

FRONTEND:
Committing Baseline Formatting for Overall NBA Statistics Page: #9 
Committing JavaScript Code for Fetching NBA API on Overall NBA Statistics Page & Fantasy Rankings Page: #10 

BACKEND: 
Committing Progress on Java Statistical Analysis Algorithm for Fantasy Points Prediction: #11 

## Nikhil
**FEATURE WORK ON FRONTEND & BACKEND**:
- I will focus on developing machine learning and AI models for team prediction analysis. These models will utilize historical data to forecast team performances and outcomes.
- On the frontend, I will be responsible for designing the homepage. This will include implementing advanced styling techniques using SASS for a more sophisticated and dynamic user interface.

### Example Machine Learning Analysis Converted to Java

```java
import weka.core.Instances;
import weka.classifiers.functions.LinearRegression;


// Other necessary imports for data manipulation and ML

public class NBAAnalysis {

    public static String fetchPlayerData(String playerName) {
        // Implementation to fetch player ID based on player name
        // This could be done using an API call or a database query
    }

    public static Instances fetchGameLogs(String playerId, String teamAbbreviation) {
        // Fetch game logs for the player against a specific team
        // Data manipulation similar to the pandas DataFrame
    }

    public static LinearRegression trainModel(Instances data) {
        // Train a linear regression model on the given data
        // Feature selection and model training 
    }

    public static double predictPoints(LinearRegression model, double[] nextGameFeatures) {
        // Predict points for the next game using the trained model
      
    }

    public static void main(String[] args) {

    }

    public static double[] estimateNextGameFeatures(Instances playerData, int recentGamesCount) {
        // Calculate averages of the last 'recentGamesCount' games

    }
}
```
Key Commits:
- HomePage Frontend Finished With all the Styling
- Fetch ML Data from backend and display output to frontend
- Different ML Models committed

### What I want to do:
I want to use an AI model to analyze previous data from players to predict their next game's performance. This would be useful to help predict games and help users make sports bets on players. The model will analyze their matchups and previous times playing the opposing team.
### Key Commits
- Frontend: Committing an analyze button on either the player's name or their page.
- Backend: Committing the analyzation model to predict the player's next game statistics.

### What I want to do:
Implement Fantasy with NBA [API](https://github.com/fparra20/Java-NBA-API), has only 2016 stats, using [jquery and DOM](https://pitsco.github.io/personal/c4.0/2024/01/08/Extra.html) table.

I would also like to work on each individual player's profile as some [key commits](https://github.com/Pitsco/nba-analysis/issues/12).

[Reference](https://fantasy.espn.com/basketball/team?leagueId=99370905&teamId=3&seasonId=2024) to what I want to implement

### Old API blogs about API implementation:
- [NBA API Github](https://github.com/fparra20/Java-NBA-API):
   - [Specific Player Stats](https://github.com/swar/nba_api/blob/master/docs/examples/Basics.ipynb)
   - [Live Game Data](https://github.com/swar/nba_api/blob/master/docs/examples/LiveData.ipynb)
      - Can be used to predict game points with fantasy

# Vishnu

## MLOps Focus:

Feature Work:
- Development of a Neural Network Framework: Crafting an advanced neural network framework using Spring Java, optimized for high-performance computations and scalability.
- Implementation of Binary Cross-Entropy and Grid Search Tuning: Employing binary cross-entropy for loss calculation and grid search for hyperparameter tuning to enhance the accuracy of NBA game predictions.
- Designing an intuitive UI component to display neural network predictions. Visual aids like graphs or heatmaps to represent data points considered by the model.

Planned Key Commits:

- Commit 1: Initial setup of the neural network framework in Spring Java.
- Commit 2: Implementation of grid search for hyperparameter optimization.
- Commit 3: Frontend development, heatmap visualization.

# Jishnu

## What I planning on doing:

- I want to devlop a user friendly site that enables for the storage and recollection of data that the user recalls from their previous experiences and is stored in the backend where it can be called every session
- Possibly implement the AI model to help recommend similar results for stats or players or maybe push recommendations for things like game predictions
- Working with other features to help make their functions more visual and enable for a better overall visualization of all the data that is passed around and add them a part of the backend data storage so it can be recalled for future experiences

Planned Key Commits:

- Commit 1: Set up working login and signup page with GET and READ working
- Commit 2: Create a profile tab that can update and delete what the user wants to update and delete
- Commit 3: Allow to use the login to store important user details and infromation (maybe implement a hashmap to store fantasy details and e.t.c.)
- Commit 4: Store and report previous predicitons for games and results vs. prediction (maybe)

**Comment by VishnuAravind12** [comment]
# Vishnu

## MLOps Focus:

Feature Work:
- Development of a Neural Network Framework: Crafting an advanced neural network framework using Spring Java, optimized for high-performance computations and scalability.
- Implementation of Binary Cross-Entropy and Grid Search Tuning: Employing binary cross-entropy for loss calculation and grid search for hyperparameter tuning to enhance the accuracy of NBA game predictions.
- Designing an intuitive UI component to display neural network predictions. Visual aids like graphs or heatmaps to represent data points considered by the model.

Planned Key Commits:

- Commit 1: Initial setup of the neural network framework in Spring Java.
- Commit 2: Implementation of grid search for hyperparameter optimization.
- Commit 3: Frontend development, heatmap visualization.
